toolchain/sem_ir/facet_type_info.h

```diff
@@ -11,6 +11,29 @@
 
 namespace Carbon::SemIR {
 
+struct SpecificInterface {
+  InterfaceId interface_id;
+  SpecificId specific_id;
+
+  static const SpecificInterface None;
+
+  friend auto operator==(const SpecificInterface& lhs,
+                         const SpecificInterface& rhs) -> bool {
+    return lhs.interface_id == rhs.interface_id &&
+           lhs.specific_id == rhs.specific_id;
+  }
+  // Canonically ordered by the numerical ids.
+  friend auto operator<=>(const SpecificInterface& lhs,
+                          const SpecificInterface& rhs)
+      -> std::strong_ordering {
+    return std::tie(lhs.interface_id.index, lhs.specific_id.index) <=>
```

Comment:
I think you might just want `= default` for the body of both of these functions.

Reply:
Should these have an `<=>` at all? I worry that it'd be an attractive nuisance, and it seems inconsistent with our `*Id` types to provide it. If we're using it somewhere (eg for canonicalization), can we use a custom comparator instead, to make it clear that we're using an arbitrary order instead of a meaningful one?

Reply:
Fixed for this and rewrite constraints as well.